# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/HostWebGLContext.cpp
# Commit: bbfe23c61add
# Full Hash: bbfe23c61add07377daf80696fd6996c0463f5b6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 8fcfac450ecc -r bbfe23c61add dom/canvas/HostWebGLContext.cpp
--- a/dom/canvas/HostWebGLContext.cpp	Mon Jun 08 20:13:01 2020 +0000
+++ b/dom/canvas/HostWebGLContext.cpp	Mon Jun 08 20:34:15 2020 +0000
@@ -7,7 +7,7 @@
 
 #include "CompositableHost.h"
 #include "mozilla/layers/LayerTransactionChild.h"
-#include "mozilla/layers/TextureClientSharedSurface.h"
+#include "mozilla/layers/LayersSurfaces.h"
 
 #include "MozFramebuffer.h"
 #include "TexUnpackBlob.h"
@@ -106,9 +106,9 @@
   (void)mOwnerData.outOfProcess->mParent.SendJsWarning(text);
 }
 
-RefPtr<layers::SharedSurfaceTextureClient> HostWebGLContext::GetVRFrame(
-    ObjectId id) const {
-  return mContext->GetVRFrame(AutoResolve(id));
+Maybe<layers::SurfaceDescriptor> HostWebGLContext::GetFrontBuffer(
+    const layers::TextureType t) const {
+  return mContext->GetFrontBuffer(t);
 }
 
 //////////////////////////////////////////////